pkg/collector/py/loader.go

```diff
@@ -97,6 +97,45 @@ func (cl *PythonCheckLoader) Load(config integration.Config) ([]check.Check, err
 
 	// Try to find a class inheriting from AgentCheck within the module
 	checkClass, err := findSubclassOf(cl.agentCheckClass, checkModule, glock)
+
+	wheelVersion := ""
+	if err == nil {
+		// getting the wheel version fo the check
+		wheelVersionPy := checkModule.GetAttrString("__version__")
+		if wheelVersionPy != nil {
+			defer wheelVersionPy.DecRef()
+			if python.PyString_Check(wheelVersionPy) {
+				wheelVersion = python.PyString_AS_STRING(wheelVersionPy.Str())
+			} else {
+				// This should never happen. If the check is a custom one
+				// (a simple .py file dropped in the check.d folder) it does
+				// not have a '__version__' attribute. If it's a datadog wheel
+				// the '__version__' is a string.
+				//
+				// If we end up here: we're dealing with a custom wheel from
+				// the user or a buggy official wheels.
+				//
+				// In any case we'll try to detect the type of '__version__' to
+				// display a meaningful error message.
+
+				typeName := "unable to detect type"
+				pyType := wheelVersionPy.Type()
+				if pyType != nil {
+					defer pyType.DecRef()
+					pyTypeStr := pyType.Str()
+					if pyTypeStr != nil {
+						defer pyTypeStr.DecRef()
+						typeName = python.PyString_AS_STRING(pyType.Str())
```

Comment:
I think we should do:
```
typeName = python.PyString_AS_STRING(pyTypeStr)
```

Otherwise we'll be creating a new reference after the `pyType.Str()` and never decreasing it. Beside, we already have the string object in `pyTypeStr`.

Reply:
:man_facepalming: ...